app/jobs/scheduled_job.rb

```diff
@@ -1,34 +1,86 @@
 class ScheduledJob < ApplicationJob
+  class SchedulerError < StandardError
+    attr_reader :original
+
+    def initialize(message = nil, original: nil)
+      @original = original
+      super(message)
+    end
+  end
+
+  class RetryableError < SchedulerError; end
+  class TooManyRetriesError < SchedulerError; end
+
   queue_as :default
 
   def perform
-    now = DateTime.now
-    beginning_of_hour = now.beginning_of_hour
-    end_of_hour       = now.end_of_hour
+    return if tasks.empty?
 
-    tasks = Scheduler.where(run_on: beginning_of_hour..end_of_hour)
+    tasks.each do |task|
+      run_task(task)
 
-    return unless tasks.size.positive?
+    rescue RetryableError => e
+      Rails.logger.warn("Task #{task.id} failed (attempt ##{task.attempts + 1}) with retryable error, rescheduling...")
+      report_rescheduled(task, e.original)
+      task.reschedule!
 
-    tasks.each do |task|
-      vendor_module = "#{task.integration.vendor.camelize}Service::Batch".constantize
-      ctx = {
-        id: nil,
-        attributes: nil,
-        relationships: {
-          batch: {
-            data: {
-              id: task.batch_id
-            }
-          },
-          facility: {
-            data: {
-              id: task.facility_id
-            }
+    rescue TooManyRetriesError => e
```

Comment:
Is there a limit on retries? If not, let's have a hard limit of 4. Please. After that, CS should contact the customer and let them know this.

Reply:
nvm